spec/controllers/controller_spec.rb

```diff
@@ -64,7 +64,13 @@
           expect(session[:user_id]).to eq user.id.to_s
         end
 
+        # NOTE: The lack of a CSRF token may mean that sessions will break
+        #       horribly for Sorcery when using Rails 7.1+. We shall see.
         it 'sets csrf token in session' do
+          if Gem::Version.new(Rails.version) >= Gem::Version.new('7.1')
+            pending 'Rails 7.1 is not including the csrf token in the session for unknown reasons'
+          end
```

Comment:
@joshbuker Sorcery uses the [form_authenticity_token method](https://github.com/Sorcery/sorcery/blob/856eacd9912ecbb436a7259ba927f171d77b0ba1/lib/sorcery/controller.rb#L57) to assign values to `session[:_csrf_token]`. However, it seems that due to [this commit](https://github.com/rails/rails/commit/f2c66ce392016efd49816cd6fdc9f174271a0321) in Rails 7.1, the behavior of the form_authenticity_token method has changed and it no longer assigns values to `session[:_csrf_token]`.

If it's necessary to assign a value to `session[:_csrf_token]` during the execution of the login method, then we may need to revise our implementation. The question arises whether this is actually necessary. Rails seems to assign values to `session[:_csrf_token]` as needed, so there might not be a clear need for Sorcery to explicitly do this. There doesn't seem to be a specific reason mentioned in the commit where [form_authenticity_token was introduced](https://github.com/Sorcery/sorcery/commit/8a7490c3b6b8e64c9b6d092103a74be77e4093e6), making it hard to infer the rationale behind this implementation.

Reply:
@joshbuker I did some additional research on this topic. The PR that introduced the form_authenticity_token method is [here](https://github.com/NoamB/sorcery/pull/409), and it seems it was merged to fix [this issue](https://github.com/NoamB/sorcery/issues/250) (though it's mentioned as `Fix for #150` in the PR, it seems like a typo and should probably be `Fix for #250` based on the content).

However, calling form_authenticity_token during login doesn't resolve the issue mentioned in https://github.com/NoamB/sorcery/issues/250, making [this commit](https://github.com/Sorcery/sorcery/commit/8a7490c3b6b8e64c9b6d092103a74be77e4093e6) seem rather pointless. Moreover, invoking form_authenticity_token has led to a new issue being created, as seen in [Using sorcery in an API-only rails app does not work · Issue #330 · Sorcery/sorcery](https://github.com/Sorcery/sorcery/issues/330), suggesting that it might be better to remove this implementation and related tests.